src/httpclient.c

```diff
@@ -671,6 +675,12 @@ int main(int argc, char **argv)
             usage(argv[0]);
             exit(0);
             break;
+        case 0:
+            if (is_opt_max_udp_payload_size == 1) {
+                h3ctx.quic.initial_egress_max_udp_payload_size = (uint16_t)strtoul(optarg, NULL, 10);
+                h3ctx.quic.transport_params.max_udp_payload_size = (uint64_t)strtoull(optarg, NULL, 10);
```

Comment:
I think I have three comments:
* Do we need to change `initial_egress_max_udp_payload_size` as well? Considering the fact that the default values of the two parameters are different, it is a bit surprising if `--max-udp-payload-size` set both to the specified value. If there's a need to set `initial_egress_max_udp_payload_size`, maybe create another option?
* For consistency, I think we might want to use `sscanf` rather than `strtoul`, to check for some usage errors at least. It has its own quirks though.
* It's long time since I last used `getopt_long` and I could be wrong, but do we need to clear `is_opt_max_udp_payload_size`, so that we do not misidentify the longopt being used when other longopts are used? This is a theoretical point at the moment as we have only one longopt, though.

Reply:
Thank you for your review @kazuho.

> * Do we need to change `initial_egress_max_udp_payload_size` as well?

I think we need it because otherwise the size of the UDP packets will still be equal to [DEFAULT_INITIAL_EGRESS_MAX_UDP_PAYLOAD_SIZE](https://github.com/h2o/quicly/blob/345b8e16d5a323b1568b11318ab7992be0be740f/lib/defaults.c#L25), regardless of what either side's `max_udp_payload_size` transport parameter is, since the client's initial message dictates the actual size of the packets as we discussed in [\[h2o/quic\] #453](https://github.com/h2o/quicly/pull/453).  I have now made it a separate option.

I have incorporated your feedback in [6cc4df0f](https://github.com/h2o/h2o/pull/2715/commits/6cc4df0ff65e28f448f987c1a07519c6f20f2b5d).